environs/__init__.py

```diff
@@ -270,29 +270,70 @@ def read_env(
         file is found. If you do not wish to recurse up the tree, you may pass
         False as a second positional argument.
         """
-        # By default, start search from the same file this function is called
-        if path is None:
+        # declare constants
+        DEFAULT_ENVIRONMENT_FILE_NAME = ".env"
+        PATH_SEPARATOR_PATTERN = r"\\|\/"
+
+        # remember original path
+        original_path = path
+
+        # use .env as path if path is None or empty
+        if not path:
+            path = DEFAULT_ENVIRONMENT_FILE_NAME
+
+        # make path absolute if it is relative, using the directory of the calling script as base
+        if not os.path.isabs(path):
+            def merge_absolute_and_relative_path(absolute_path, relative_path):
+                absolute_path_items = re.split(PATH_SEPARATOR_PATTERN, absolute_path)
+                relative_path_items = re.split(PATH_SEPARATOR_PATTERN, relative_path)
+                popped = False
+                for absolute_path_item in absolute_path_items:
+                    if absolute_path_item == relative_path_items[0]:
+                        relative_path_items.pop(0)
+                        popped = True
+                    elif popped:
+                        break
+                return "/".join(absolute_path_items + relative_path_items)
```

Comment:
This won't work on non-*nix systems that don't use `/` as the path separator. You shouldn't need to worry about path separators in Python. `os.path` and `pathlib` both have ways to deal with them transparently.

Reply:
Actually, I ran your unit tests on a Windows machine and did not have issues with / as path separators. However, I agree using os.path.sep is better.